barretenberg/cpp/src/barretenberg/stdlib_circuit_builders/ultra_circuit_builder.hpp

```diff
@@ -421,7 +421,7 @@ class UltraCircuitBuilder_ : public CircuitBuilderBase<typename Arithmetization_
 #endif // NDEBUG
     }
 
-    void finalize_circuit();
+    void finalize_circuit(const bool ensure_nonzero = false);
```

Comment:
I wanted to preserve the ability to finalize without ensuring nonzero because this is how the function is used in most places (e.g. in the SMT solver). I should raise this issue internally to make sure people know this is happening.

Reply:
I feel like this makes things more confusing. Like we're going to have unfinalized gate count, sorta finalized gate count without ensure_nonzero gates, and also a finalized gate count...

Reply:
Yeah I'd rather not add this but I also don't want to change underlying assumptions--literally only one case DOES ensure gates are nonzero...